modules/apigee_edge_teams/src/Plugin/EntityReferenceSelection/TeamMembersSelection.php

```diff
@@ -169,7 +170,13 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
    * {@inheritdoc}
    */
   protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS') {
-    $query = parent::buildEntityQuery($match, $match_operator);
+    // Do not pass $match so that it does not add the "name" field to the query,
+    // as we will search by email instead.
+    $query = parent::buildEntityQuery(NULL, $match_operator);
```

Comment:
As far as I see there is no better way than this only if you build your own implementation :(